lib/OpenQA/Shared/Controller/Auth.pm

```diff
@@ -49,7 +49,7 @@ sub auth ($self) {
     # Browser with a logged in user
     my ($user, $reason) = (undef, 'Not authorized');
     if ($user = $self->current_user) {
-        ($user, $reason) = (undef, 'Bad CSRF token!') unless $self->valid_csrf;
+        ($user, $reason) = (undef, 'Bad CSRF token!') unless $self->req->method eq 'GET' || $self->valid_csrf;
```

Comment:
Why do you need this change? there is a comment in the valid_csrf which made me think that if this restriction has to be explicit it has to be better in the valid_csrf, no?

Reply:
Your first question is answered in the last paragraph of the commit message of the first commit.

I don't understand the second question completely. I can only say that from a security standpoint is is of course generally better to be restrictive. However, I suppose here we have to make a change to be less restrictive. The reasoning why I think this is fine is also explained in the commit message.